### PR TITLE
test(RHINENG-21087): Add E2E test to rename and delete workspace from  Workpsaces page

### PR DESCRIPTION
**Pull request by computercamplove**
1. Add E2E test to rename/delete workspace from main Workspaces page
2. Add` workspaceHelper.ts `with repeated actions
3. Add docstring to function to not repeat comments
4. Rename `navigateToInventoryFunc()` navigation to `navigateToInventorySystemsFunc()` -  use the most specific name possible that still accurately describes the function's responsibility.
5. Fix-up conventional style for variables

## Summary by Sourcery

Add and refactor workspace-related E2E tests by introducing reusable helpers, renaming navigation utilities, and covering negative delete scenarios for workspaces with systems

New Features:
- Add E2E test ensuring a workspace with attached systems cannot be deleted

Enhancements:
- Introduce workspaceHelpers with generateUniqueWorkspaceName and createNewWorkspace utilities
- Refactor existing workspace test to leverage helper functions and streamline naming and search flows
- Rename navigateToInventoryFunc to navigateToInventorySystemsFunc and update all references
- Add JSDoc comments to navigation and popup helper functions

Tests:
- Restructure workspace CRUD tests into discrete test.step blocks with improved modal and heading assertions

## Summary by Sourcery

Introduce reusable workspace test helpers, rename navigation utility, and expand E2E coverage for workspace actions

New Features:
- Add E2E test preventing deletion of workspaces with attached systems
- Add E2E test for renaming and deleting workspaces directly from the Workspaces page

Enhancements:
- Introduce workspaceHelpers.ts with generateUniqueWorkspaceName and createNewWorkspace utilities
- Refactor workspace tests to leverage helper functions, test.step blocks, and consistent naming conventions
- Rename navigateToInventoryFunc to navigateToInventorySystemsFunc and update all references
- Add JSDoc comments to navigation and popup helper functions

Documentation:
- Add docstrings to reduce duplicated inline comments in helper modules

Tests:
- Improve modal and heading assertions and split workspace tests into granular steps
- Update navigation and ungrouped workspace tests to use the renamed inventory navigation function

### CODE REVIEW

File: _playwright-tests/test_workspace.test.ts [hunk 1]
```diff
@@ -2,8 +2,14 @@ import { test, expect } from '@playwright/test';
 
 import { navigateToWorkspacesFunc } from './helpers/navHelpers';
 import { closePopupsIfExist } from './helpers/loginHelpers';
+import { generateUniqueWorkspaceName, createNewWorkspace } from './helpers/workspaceHelpers';
+import { searchByName } from './helpers/filterHelpers';
 
-test('User can create, rename, and delete a workspace', async ({ page }) => {
+test.beforeEach(async ({ page }) => {
```
Comment (adarshdubey-star) [inline _playwright-tests/test_workspace.test.ts:8]: Really like this 👍

Comment (adarshdubey-star) [reply _playwright-tests/test_workspace.test.ts:8]: Implemented in another [PR](https://github.com/RedHatInsights/insights-inventory-frontend/pull/2653/files#diff-1b200afc7fce124eb696dc2fa69b19ac19f8dd496f324062db7245e493e46c1eR14)

**Review by adarshdubey-star — APPROVED** [review]
Very well implemented @computercamplove 👍

File: _playwright-tests/test_workspace.test.ts [hunk 2]
```diff
@@ -168,9 +170,71 @@ test('User able to bulk delete empty workspaces', async ({ page }) => {
...
+  const renamedWorkspace = `${workspaceName}_Renamed`;
+  const dialogModal = page.locator('[data-ouia-component-id="group-modal"]');
+  const perRowKebabButton = page.getByRole('button', { name: 'Kebab toggle' });
+  const perRowMenu = page.locator('[class="pf-v6-c-menu"]'); 
+  const nameColumnLocator = page.locator('td[data-label="Name"]');
+
+  await test.step('Test setup: navigate to Workspaces page, create workspace to work with', async () => {
+    await navigateToWorkspacesFunc(page);
+    await createNewWorkspace(page, workspaceName);
+    // serach for workspace and via 'Name' column make sure only 1 workspace is found
```
[the remark this answers is not in the document]
Comment (adarshdubey-star) [reply _playwright-tests/test_workspace.test.ts:205]: I will update this in my PR no need to commit again.
